src/objects/zcl_abapgit_object_iaxu.clas.abap

```diff
@@ -0,0 +1,305 @@
+
+    METHODS:
+      read
+        EXPORTING es_attr TYPE w3tempattr
+        RAISING   zcx_abapgit_exception,
+      save
+        IMPORTING is_attr TYPE w3tempattr
+        RAISING   zcx_abapgit_exception,
+      w3_api_load
+        IMPORTING is_name    TYPE iacikeyt
+        EXPORTING eo_xml_api TYPE REF TO cl_w3_api_xml3
+                  es_attr    TYPE w3tempattr
+        RAISING   zcx_abapgit_exception,
+      w3_api_set_changeable
+        IMPORTING io_xml_api    TYPE REF TO cl_w3_api_xml3
+                  iv_changeable TYPE abap_bool
+        RAISING   zcx_abapgit_exception,
+      w3_api_delete
+        IMPORTING io_xml_api TYPE REF TO cl_w3_api_xml3
+        RAISING   zcx_abapgit_exception,
+      w3_api_save
+        IMPORTING io_xml_api TYPE REF TO cl_w3_api_xml3
+        RAISING   zcx_abapgit_exception,
+      w3_api_create_new
+        IMPORTING is_attr    TYPE w3tempattr
+        EXPORTING eo_xml_api TYPE REF TO cl_w3_api_xml3
+        RAISING   zcx_abapgit_exception.
+
+ENDCLASS.
+
+
+
+CLASS zcl_abapgit_object_iaxu IMPLEMENTATION.
+
+
+  METHOD read.
+
+    DATA: ls_name TYPE iacikeyt.
+
+
+    ls_name = ms_item-obj_name.
+
+    w3_api_load( EXPORTING is_name = ls_name
+                 IMPORTING es_attr = es_attr ).
+
+    CLEAR: es_attr-chname,
+           es_attr-tdate,
+           es_attr-ttime,
+           es_attr-devclass.
+
+  ENDMETHOD.
+
+
+  METHOD save.
+
+    DATA: lo_xml_api TYPE REF TO cl_w3_api_xml3.
+
+
+    w3_api_create_new( EXPORTING is_attr    = is_attr
+                       IMPORTING eo_xml_api = lo_xml_api ).
+
+    w3_api_save( io_xml_api = lo_xml_api ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~changed_by.
+    rv_user = c_user_unknown. " todo
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~delete.
+
+    DATA: lo_xml_api TYPE REF TO cl_w3_api_xml3,
+          ls_name    TYPE iacikeyt,
+          lx_root    TYPE REF TO zcx_abapgit_exception.
+
+
+    ls_name = ms_item-obj_name.
+
+    w3_api_load( EXPORTING is_name    = ls_name
+                 IMPORTING eo_xml_api = lo_xml_api ).
+
+    w3_api_set_changeable( io_xml_api    = lo_xml_api
+                           iv_changeable = abap_true ).
+
+    w3_api_delete( io_xml_api = lo_xml_api ).
+
+    w3_api_save( io_xml_api = lo_xml_api ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~deserialize.
+
+    DATA: ls_attr TYPE w3tempattr.
+
+
+    io_xml->read( EXPORTING iv_name = 'ATTR'
+                  CHANGING  cg_data = ls_attr ).
+
+    ls_attr-devclass = iv_package.
+
+    IF zif_abapgit_object~exists( ) = abap_true.
+      zif_abapgit_object~delete( ).
+    ENDIF.
+
+    save( is_attr = ls_attr ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~exists.
+
+    DATA: ls_name TYPE iacikeyt.
+
+
+    ls_name = ms_item-obj_name.
+
+    TRY.
+        w3_api_load( is_name = ls_name ).
+
+        rv_bool = abap_true.
+
+      CATCH zcx_abapgit_object_not_found.
+        rv_bool = abap_false.
+
+    ENDTRY.
```

Comment:
Can't import test repo
![image](https://user-images.githubusercontent.com/17437789/67237334-55293800-f44b-11e9-9830-eabafbe15040.png)

`zcx_abapgit_not_found=>raise` raises exception of type zcx_abapgit_exception  w3_api_load which isn't caught here.

Reply:
Fixed it